wom/src/test/scala/wdl4s/wom/graph/CallNodeSpec.scala

```diff
@@ -0,0 +1,123 @@
+      "declaration_satisfied_by_port" -> declarationPort
+    )
+
+    val requiredExpression = GraphNodeInputExpression("required_satisfied_by_expression", PlaceholderWomExpression(Set.empty, WdlStringType), Map("a.out" -> GraphNodeOutputPort("a", WdlStringType, null)))
+    val optionalExpression = GraphNodeInputExpression("optional_satisfied_by_expression", PlaceholderWomExpression(Set.empty, WdlStringType), Map("a.out" -> GraphNodeOutputPort("a", WdlStringType, null)))
+    val optionalWithDefaultExpression = GraphNodeInputExpression("optional_with_default_satisfied_by_expression", PlaceholderWomExpression(Set.empty, WdlStringType), Map("a.out" -> GraphNodeOutputPort("a", WdlStringType, null)))
+    val declarationExpression = GraphNodeInputExpression("declaration_satisfied_by_expression", PlaceholderWomExpression(Set.empty, WdlStringType), Map("a.out" -> GraphNodeOutputPort("a", WdlStringType, null)))
+
+    val expressionInputs = Set(requiredExpression, optionalExpression, optionalWithDefaultExpression, declarationExpression)
+
+    val callWithInputs = CallNode.callWithInputs("call_name", taskDefinition, portInputs, expressionInputs).getOrElse(fail("Failed to create call with inputs"))
+
+    val newInputs = callWithInputs.newInputs
+    
+    // One for each unsatisfied ExternalGraphInputNode (DeclaredInputDefinitions immediately fall back to their default and 
+    // don't generate graph input nodes)
+    newInputs should have size 3
+
+    val requiredInput = newInputs.find(_.name == "call_name.required_unsatisfied")
+      .getOrElse(ginNotFound("call_name.required_unsatisfied"))
+    requiredInput.isInstanceOf[RequiredGraphInputNode] shouldBe true
+
+    val optionalInput = newInputs.find(_.name == "call_name.optional_unsatisfied")
+      .getOrElse(ginNotFound("call_name.optional_unsatisfied"))
+    optionalInput
+      .isInstanceOf[OptionalGraphInputNode] shouldBe true
+
+    val optionalInputWithDefault = newInputs.find(_.name == "call_name.optional_with_default_unsatisfied")
+      .getOrElse(ginNotFound("call_name.optional_with_default_unsatisfied"))
+    optionalInputWithDefault
+      .asInstanceOf[OptionalGraphInputNodeWithDefault].default shouldBe optionalWithDefaultUnsatisfied.default
+
+    // Validate input mappings are correct
+    val inputDefinitionMappings = callWithInputs.node.inputDefinitionMappings
+    inputDefinitionMappings.size == taskDefinition.inputs.size shouldBe true
+
+    /* *** Unsatisfied *** */
+    // unsatisfied input defs should map to the above newInput output ports
+    inputDefinitionMappings(requiredUnsatisfied) eq requiredInput.singleOutputPort shouldBe true
+    inputDefinitionMappings(optionalUnsatisfied) eq optionalInput.singleOutputPort shouldBe true
+    inputDefinitionMappings(optionalWithDefaultUnsatisfied) eq optionalInputWithDefault.singleOutputPort shouldBe true
+    
+    // unsatisfied declared input def gets a FloatingExpressionNode
```

Comment:
FloatingExpressionNode!

Reply:
argh I renamed the class but not the comment